packages/browser/examples/demoClientApp/src/DemoClientApp.js

```diff
@@ -212,9 +212,22 @@ class DemoClientApp extends Component {
     // this.lookupIdentityProviderConfig(this.state.loginIssuer);
   }
 
-  highlightPartOfWebId(webId, part) {
-    // Example: https://ldp.demo-ess.inrupt.com/110592712913443799002/profile/card#me
+  /**
+   * Very hacky attempt at splitting a WebID into 'components', such that:
+   *  Component 0 is up to, but not including, the username part of the WebID.
+   *  Component 1 is just the username part of the WebID.
+   *  Component 2 is everything after the username part of the WebID.
+   *
+   * Example: https://ldp.demo-ess.inrupt.com/110592712913443799002/profile/card#me
+   *  Component 0 is "https://ldp.demo-ess.inrupt.com/".
+   *  Component 1 is "110592712913443799002".
+   *  Component 2 is "/profile/card#me".
+   */
+  extractComponentOfWebId(webId, part) {
```

Comment:
Now that I think about it, don't we want to discourage from doing this kind of processing in the IRI at all, and consider it opaque ? I know that foaf isn't great, but shouldn't we look up the user profile for their `foaf:nick` or `foaf/vcard:name` instead ?

Reply:
Yeah, for sure (which is why the comment says this is 'very hacky'). It's an interesting question though, which I'd basically articulate as '**_WebID's are the core identifier, but they're also kinda terrible because they're known to be terrible identifiers for humans to remember (based on Yahoo research from many years ago), and so can we (or should we even try to) easily render the 'important', human-friendly portion of a WebID without having to dereference and lookup the profile document?_**'.
Here, I specifically **_don't_** want to display the `foaf:nick`, or the user's `rdfs:label` or `rdfs:comment`, or something - I specifically want to highlight the differences between different WebID's, if possible. We can do this currently with out-of-the-box ESS and NSS generated WebID's, but that's only 'cos they follow particular naming conventions (e.g. Sarven's WebID doesn't follow either: https://csarven.ca#i).
But yeah, I think I'll probably remove this whole thing, as we'll never be able to reliably determine anything from WebIDs. It was useful when I was first demo'ing this app, 'cos I was using two WebID's from the the same ESS server, which made it really difficult for a human to visually distinguish between them.
But perhaps I could replace this with a simpler, far more generic 'visually highlight the differences between two WebID's', as I think that could be implemented with simple string comparison (i.e. if the two strings only differ by a single substring, then just highlight that substring in both, or something like that...). I'll create a little note-to-self backlog ticket: